src/Microsoft.TestPlatform.Common/Utilities/TypesToLoadUtilities.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright(c) Microsoft Corporation.All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.Common.Utilities
+{
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel.Utilities;
+
+    using System;
+    using System.Collections.Generic;
+    using System.Reflection;
+    using System.Linq;
+
+    internal static class TypesToLoadUtilities
+    {
+        public const string TypesToLoadAttributeFullName = "Microsoft.VisualStudio.TestPlatform.ObjectModel.Utilities.TypesToLoadAttribute";
```

Comment:
You have several `TypesToLoadAttribute` classes, but here loading the one which is in `ObjectModel`. How does it work for other types?

Reply:
Public assembly-level attribute causing compatibility issues with older versions of VS, where that attribute is not present. We removed the attribute from `Microsoft.VisualStudio.TestPlatform.ObjectModel` but kept the functionality by allowing extensions to define that internally. 

We look for an attribute with the full name of `Microsoft.VisualStudio.TestPlatform.ObjectModel.Utilities.TypesToLoadAttribute` defined in target assembly. If we find it and it has a property called `Types`, which returns a `System.Type` array, we use that.

Reply:
Yes, I see it, I got you created several `TypesToLoadAttribute` classes in `Microsoft.VisualStudio.TestPlatform.ObjectModel.Utilities.TypesToLoadAttribute` namespace.
But the main question : how will it help to solve the main problem in MSTestAdapter?

Reply:
The same way I did here. Check this https://github.com/microsoft/testfx/compare/master...Haplois:fix/removed-types-to-load.